### PR TITLE
feat(book/src/languages.md)

### PR DESCRIPTION
**Pull request by ammkrn**
Add a section in the book about language-specific settings and the languages.toml file.

### CODE REVIEW

File: book/src/languages.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Languages
+
+Language-specific settings and settings for particular language servers can be configured in a `languages.toml` file placed in your configuration directory (e.g. `~/.config/helix/languages.toml`). An example which includes the defaults used by helix can be found in the main helix repository's [languages.toml](https://github.com/helix-editor/helix/blob/master/languages.toml) file.
```
Comment (sudormrfbin) [inline book/src/languages.md:3]:
```suggestion
Language-specific settings and settings for particular language servers can be configured in a `languages.toml` file placed in your [configuration directory](./configuration.md). An example which includes the defaults used by helix can be found in the main helix repository's [languages.toml](https://github.com/helix-editor/helix/blob/master/languages.toml) file.
```

**Comment by archseer** [comment]
Would you mind merging in some of the text from https://github.com/helix-editor/helix/pull/710 ?

**Comment by ammkrn** [comment]
> Would you mind merging in some of the text from #710 ?

@archseer It doesn't look like the question about what `roots` is for was ever resolved. I only added the configuration page because I couldn't figure out how to turn off auto-format and language configuration wasn't in the docs. Adding a new language to the compile-time toml file seems more involved, I'll probably put it in a separate section.

**Comment by ammkrn** [comment]
@archseer `adding_languages` is now included, as well as the suggested changes from the #710 discussion. Someone's still on the hook for a description of `roots`.

File: book/src/languages.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+roots = []
+auto-format = false
+comment-token = "//"
+language-server = { command = "rust-analyzer" }
+indent = { tab-width = 4, unit = "    " }
+[language.config]
+cargo = { loadOutDirsFromCheck = true }
+procMacro = { enable = false }
+```
+
```
Comment (archseer) [inline book/src/languages.md:24]: Actually, the configs get merged with defaults now so it's enough only override `auto-format`:

```toml
[[language]]
name = "rust"
auto-format = false
```

File: book/src/guides/adding_languages.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+These are the available keys and descriptions for the file.
+
+| Key           | Description                                                   |
+| ----          | -----------                                                   |
+| name          | The name of the language                                      |
+| scope         | A string like `source.js` that identifies the language. Currently, we strive to match the scope names used by popular TextMate grammars and by the Linguist library. Usually `source.<name>` or `text.<name>` in case of markup languages |
+| injection-regex | regex pattern that will be tested against a language name in order to determine whether this language should be used for a potential language injection site. [link](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection) |
+| file-types    | The filetypes of the language, for example `["yml", "yaml"]`  |
+| roots         | description TBD                                               |
```
Comment (archseer) [inline book/src/guides/adding_languages.md:32]:
```suggestion
| roots         | A set of marker files to look for when trying to find the workspace root. For example `Cargo.lock`, `yarn.lock` |
```

File: book/src/guides/adding_languages.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+| auto-format   | Whether to autoformat this language when saving               |
+| comment-token | The token to use as a comment-token                           |
+| indent        | The indent to use. Has sub keys tab-width and unit            |
```
Comment (archseer) [inline book/src/guides/adding_languages.md:35]:
```suggestion
| indent        | The indent to use. Has sub keys `tab-width` and `unit`            |
```

**Review by archseer — CHANGES_REQUESTED** [review]
Looks great! Two small tweaks:

**Comment by ammkrn** [comment]
@archseer 

The formatting suggestions are incorporated and the languages.md page was rewritten to mention the merge behavior as well as the role of both languages.toml files.

**Comment by archseer** [comment]
Thanks, documentation improvements are always great! 🎉

**Pull request merged**